server.js

```diff
@@ -1966,6 +1966,50 @@ cache(function(data, match, sendBadge, request) {
   });
 }));
 
+// GitHub release-download-count integration.
+camp.route(/^\/github\/downloads\/([^\/]+)\/([^\/]+)\/([^\/]+)\/([^\/]+)\.(svg|png|gif|jpg|json)$/,
+cache(function(data, match, sendBadge, request) {
+  var user = match[1];  // eg, qubyte/rubidium
+  var repo = match[2];
+  var id = match[3];
+  var asset_name = match[4].toLowerCase();
+  var format = match[5];
+  var apiUrl = 'https://api.github.com/repos/' + user + '/' + repo + '/releases/' + id;
+  // Using our OAuth App secret grants us 5000 req/hour
+  // instead of the standard 60 req/hour.
+  if (serverSecrets) {
+    apiUrl += '?client_id=' + serverSecrets.gh_client_id
+      + '&client_secret=' + serverSecrets.gh_client_secret;
+  }
+  var badgeData = getBadgeData('downloads', data);
+  // A special User-Agent is required:
+  // http://developer.github.com/v3/#user-agent-required
+  request(apiUrl, { headers: githubHeaders }, function(err, res, buffer) {
+    if (err != null) {
+      badgeData.text[1] = 'inaccessible';
+      sendBadge(format, badgeData);
+    }
+    try {
+      if ((+res.headers['x-ratelimit-remaining']) === 0) {
+        return;  // Hope for the best in the cache.
+      }
+      var data = JSON.parse(buffer);
+      var downloads = 0;
+      data.assets.map(function (asset) {
```

Comment:
That microbenchmark describes `map` and `forEach` as being roughly in the same ballpark (due to inlining work not being finished in V8), with `forEach` slightly faster. I believe you read the graph wrong.

If you'd like to make a change for performance reasons, I would suggest using a for loop with a cached length, as that is known to be the easiest to optimize for JS engines.